homeassistant/components/recorder/models.py

```diff
@@ -149,6 +152,53 @@ def to_native(self, validate_entity_id: bool = True) -> Event | None:
             return None
 
 
+class EventData(Base):  # type: ignore[misc,valid-type]
```

Comment:
Can we extract and store `device_id` from the data? Given that many remotes are using that, it would be nice to be able to easily find them. Or would that be pre-optimization and I should just do that when I add code to show them in the frontend ?

Reply:
I'll analyze the data in my instances I've concerned and see

Reply:
```
sqlite> select count(*) from event_data where shared_data like '%device_id%';
8
sqlite> select count(*) from event_data where shared_data not like '%device_id%';
969
```

I think it is probably premature at this point. If its turns out to be problem, I'll optimize for it. 😄